### ISSUE TITLE
Resw aka Resources can't be Accessed Between WinUI Based Class Library Projects in Xaml

### ISSUE THREAD

**Issue by AathifMahir** [microsoft/WindowsAppSDK#4310]
Resw aka Resources can't be Accessed Between WinUI Based Class Library Projects in Xaml.

When Trying to Access Resources that Part of Main App Project with Xaml from a WinUI Based Class Library Which is Not Directly Dependent on Main Project Instead Main Project is Depended on that WinUI Based Class Library, it just doesn't Work and No Exception Whatsoever, If we try the Same by Using ResourceLoader in C#, It Just Works as Expected, I'm not sure this a bug or it's Intended Behavior

- Repo for Reproducing
https://github.com/AathifMahir/ResWDemoAppProject

**Comment by ghost1372** [comment]
You are doing wrong...
You can not use strings exist in Main Project in class library, but you can use class library strings resources in your Main App

**Comment by AathifMahir** [comment]
> You are doing wrong... You can not use strings exist in Main Project in class library, but you can use class library strings resources in your Main App

I'm looking for way to use strings from class library, but that doesn't work at all and the app doesn't pick up any resources when the resources were in class library but as an exception when we have exactly same resources in class library and app project, that fixes the issue. I'm not sure on how resource work anymore since resources are app wide not constraints to different projects

**Comment by ghost1372** [comment]
> > You are doing wrong... You can not use strings exist in Main Project in class library, but you can use class library strings resources in your Main App
> 
> I'm looking for way to use strings from class library, but that doesn't work at all and the app doesn't pick up any resources when the resources were in class library but as an exception when we have exactly same resources in class library and app project, that fixes the issue. I'm not sure on how resource work anymore since resources are app wide not constraints to different projects, The Resources are App wide is True when we are accessing resources using ResourceLoader or ResourceMap in c# aka CodeBehind or Somewhere else. but doesn't work with Xaml
> 
> If i remember correctly, Previously on UWP having Packaging Project with Strings in it Fixes these kind of issues but Not on WinUI 3

See here
https://github.com/microsoft/WindowsAppSDK-Samples/tree/main/Samples/ResourceManagement/cs/cs-winui

**Comment by AathifMahir** [comment]
> > > You are doing wrong... You can not use strings exist in Main Project in class library, but you can use class library strings resources in your Main App
> > 
> > I'm looking for way to use strings from class library, but that doesn't work at all and the app doesn't pick up any resources when the resources were in class library but as an exception when we have exactly same resources in class library and app project, that fixes the issue. I'm not sure on how resource work anymore since resources are app wide not constraints to different projects, The Resources are App wide is True when we are accessing resources using ResourceLoader or ResourceMap in c# aka CodeBehind or Somewhere else. but doesn't work with Xaml
> > 
> > If i remember correctly, Previously on UWP having Packaging Project with Strings in it Fixes these kind of issues but Not on WinUI 3
> 
> See here
> https://github.com/microsoft/WindowsAppSDK-Samples/tree/main/Samples/ResourceManagement/cs/cs-winui

I tried example in the url, that also has same issue when using Xaml, with c# resources works perfectly but with Xaml I keep facing the same issue unless we have a resources.resw file on that assembly

**Comment by df-macom** [comment]
I am having similar issues. I have a microservices application where features are separated into their own class libraries (using the WInUI Class library project templates). I am attempting to give each project/microservice its own .resw file. Even when only trying to use the .resw file and its resources _within_ its associated class library, i am getting exceptions thrown for the resource map not existing. Have been unable to figure it out for a couple days now. Very frustrating

**Comment by df-macom** [comment]
> I am having similar issues. I have a microservices application where features are separated into their own class libraries (using the WInUI Class library project templates). I am attempting to give each project/microservice its own .resw file. Even when only trying to use the .resw file and its resources _within_ its associated class library, i am getting exceptions thrown for the resource map not existing. Have been unable to figure it out for a couple days now. Very frustrating

Of course, almost immediately after commenting and almost giving up and pursing a different route I found the fix at least for my case. Even though my microservices are accessing the internal .resw files from within the class library where they were defined, I still had to access the respective ResourceMap through the "GetSubtree" method of the ResourceManager.MainResourceMap("classlibrary_resource_file") where "classlibrary_resource_file" is the name of the class library followed by "/_ResourceFileNameWithoutExtension_". So if I have a class library named ClassLibrary and a .resw file in ClassLibrary/Folder/SubFolder/ClassLibraryResources.resw, I needed to pass "ClassLibrary/ClassLibraryResources.resw" into ResourceManager.MainResourceMap.GetSubtree(string name). This seems to be the only way I have been able to access the resource files. Not intuitive and despite a lot of documentation on resources as a whole I was not able to find this type of detail anywhere.

If anyone else comes across this with a similar issue, try this approach and let me know what you find.
